caom2/caom2/observation.py

```diff
@@ -318,22 +318,23 @@ def sequence_number(self):
 
     @sequence_number.setter
     def sequence_number(self, value):
-        caom_util.type_check(value, int, 'sequence_number')
-        self._sequence_number = value
+        caom_util.type_check(value, int_32, 'sequence_number')
+        self._sequence_number = int_32(value) if value is not None else None
+
 
     @property
-    def obs_type(self):
+    def type(self):
         """The OBSTYPE of the observation being recorded.
 
         eg. OBJECT, FLAT, BIAS
         type: unicode str
         """
-        return self._obs_type
+        return self._type
 
-    @obs_type.setter
-    def obs_type(self, value):
-        caom_util.type_check(value, str, 'obs_type')
-        self._obs_type = value
+    @type.setter
+    def type(self, value):
```

Comment:
We may want to adopt a convention when naming the getters and setters. For instance, all getters should either be type(self) or get_type(self) and all setters should either be type(self, value) or set_type(self, value). Currently we have a mix of these names.